src/timeout.ts

```diff
@@ -77,6 +74,12 @@ export class Timeout extends Promise<never> {
         // Ensure we do not keep the Node.js event loop running
         this.id.unref();
       }
+    } else if (duration < 0) {
```

Comment:
Allow for negative timeouts that time out on the next tick. When paired with a negative value of `TimoutContext.remainingTimeMS`, this allows for the instantiation of `MongoOperationError` instances at a more appropriate place in the driver to generate more useful stack traces.